src/Model/Contact.php

```diff
@@ -1686,4 +1686,56 @@ public static function pruneUnavailable(array &$contact_ids)
 
 		$contact_ids = $return;
 	}
+
+	/**
+	 * @brief Returns a magic link to authenticate remote visitors
+	 *
+	 * @param string $contact_url The address of the target contact profile
+	 * @param integer $url An url that we will be redirected to after the authentication
+	 *
+	 * @return string with "redir" link
+	 */
+	public static function magicLink($contact_url, $url = '')
+	{
+		$cid = self::getIdForURL($contact_url, 0, true);
+		if (empty($cid)) {
+			return ($url != '') ? $url : $contact_url;
```

Comment:
This can be shortened to `$url ?: $contact_url`.

Reply:
Looks weird.

Reply:
Wait until we start supporting only PHP 7, we'll use `$_POST['key'] ?? $default` instead of `defaults($_POST, 'key', $default)`

Reply:
Has it gained the first or only second prize at the obfuscated code awards?

Reply:
Less verbose, yeah that's the problem :-)

Reply:
Sometimes verbosity is obfuscation.